subsys/bluetooth/host/hci_core.h

```diff
@@ -198,3 +198,7 @@ void bt_id_del(struct bt_keys *keys);
 int bt_setup_id_addr(void);
 
 void bt_dev_show_info(void);
+
+int adv_start(const struct bt_le_adv_param *param, const struct bt_data *ad,
+	      size_t ad_len, const struct bt_data *sd, size_t sd_len,
+	      const bt_addr_le_t *peer);
```

Comment:
Why is this here? Just make it a static function in hci_core.c (move it high enough so you don't need to do forward declarations for it)

Reply:
I need to access this function in bt_conn_create_slave_le() (in the conn.c file), so it cannot be static

Reply:
Right, sorry, I forgot that conn.c has the directed advertising function. Then we should add a bit of name-spacing to the function's name. E.g. everything in hci_core.h right now is at least prefixed with bt_ so we'll want to be consistent with that. The obvious name is already taken by the public API. `bt_le_adv_start_internal()` perhaps? (not really liking the length of it)